aten/src/ATen/native/mkldnn/RNN.cpp

```diff
@@ -0,0 +1,757 @@
+#include <ATen/native/RNN.h>
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+#include <ATen/InitialTensorOptions.h>
+#include <ATen/MatrixRef.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/TensorUtils.h>
+#include <c10/util/Exception.h>
+
+#if !AT_MKLDNN_ENABLED()
+
+namespace at { namespace native {
+
+std::tuple<Tensor, Tensor, Tensor, std::vector<Tensor>> mkldnn_rnn(
+    const Tensor& input_r, TensorList weight, int64_t weight_stride0,
+    const Tensor& hx, const Tensor& cx,
+    int64_t fn_mode, int64_t fn_hidden_size,
+    int64_t fn_num_layers, bool batch_first, double fn_dropout,
+    bool fn_train, bool fn_bidirectional, IntList fn_batch_sizes,
+    const Tensor& fn_dropout_state) {
+  AT_ERROR("mkldnn_rnn: ATen not compiled with MKLDNN support");
+}
+
+std::tuple<Tensor, Tensor, Tensor, std::vector<Tensor>> mkldnn_rnn_backward(
+    const Tensor& input_r, TensorList weight, int64_t weight_stride0,
+    const Tensor& hx, const Tensor& cx, const Tensor& output_r,
+    const Tensor& grad_output_r, const Tensor& grad_hy_r, const Tensor& grad_cy_r,
+    int64_t fn_mode, int64_t fn_hidden_size,
+    int64_t fn_num_layers, bool batch_first, double fn_dropout,
+    bool fn_train, bool fn_bidirectional, IntList fn_batch_sizes,
+    const Tensor& fn_dropout_state, TensorList reserve,
+    std::array<bool, 4> output_mask) {
+  AT_ERROR("mkldnn_rnn_backward: ATen not compiled with MKLDNN support");
+}
```

Comment:
When mkldnn is not enabled, you still have to have dummy implementations of the stubs so linking works.